File: Data-Structures/Binary-Search-Tree/BinarySearch.cs
namespace Binary
{
    class BinarySeach
    {
        public static bool Search(int[] array, int value)
        {
            int lo = 0;
            int hi = array.Length - 1;

            do
            {
                int middle = (int)Math.Floor((hi + lo) / 2.0);
                if (array[middle] == value) return true;
                else if (array[middle] > value)
                {
                    hi = middle - 1;
                }
                else
                {
                    lo = middle + 1;
                }
            } while (lo <= hi);
            return false;
        }
    }
}

File: Algorithms/Search Algorithm/BinarySearch/binarySearch.lua
local function binarySearch(array, needle)
	local first = 1
	local last = #array
	while first <= last do
		local middle = math.floor((first + last) / 2)
		if array[middle] == needle then
			return middle
		elseif array[middle] < needle then
			first = middle + 1
		else
			last = middle - 1
		end
	end
	return nil
end

print(binarySearch({1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, 5))
